lock-api/src/main/java/com/palantir/lock/client/NamespacedConjureTimelockServiceImpl.java

```diff
@@ -85,12 +89,34 @@ public ConjureWaitForLocksResponse waitForLocks(ConjureLockRequest request) {
 
     @Override
     public ConjureRefreshLocksResponse refreshLocks(ConjureRefreshLocksRequest request) {
-        return conjureTimelockService.refreshLocks(AUTH_HEADER, namespace, request);
+        log.trace(
+                "Attempting to refresh locks V1",
+                SafeArg.of("namespace", namespace),
+                SafeArg.of("allLocksSize", request.getTokens().size()));
+        ConjureRefreshLocksResponse response = conjureTimelockService.refreshLocks(AUTH_HEADER, namespace, request);
+        log.trace(
+                "Finished refreshing locks V1",
+                SafeArg.of("namespace", namespace),
+                SafeArg.of("allLocksSize", request.getTokens().size()),
+                SafeArg.of("refreshedLocksSize", response.getRefreshedTokens().size()),
+                SafeArg.of("refreshedAll", request.getTokens().equals(response.getRefreshedTokens())));
+        return response;
     }
 
     @Override
     public ConjureRefreshLocksResponseV2 refreshLocksV2(ConjureRefreshLocksRequestV2 request) {
-        return conjureTimelockService.refreshLocksV2(AUTH_HEADER, namespace, request);
+        log.trace(
+                "Attempting to refresh locks V2",
+                SafeArg.of("namespace", namespace),
+                SafeArg.of("allLocksSize", request.get().size()));
+        ConjureRefreshLocksResponseV2 response = conjureTimelockService.refreshLocksV2(AUTH_HEADER, namespace, request);
+        log.trace(
+                "Finished refreshing locks V2",
+                SafeArg.of("namespace", namespace),
+                SafeArg.of("allLocksSize", request.get().size()),
+                SafeArg.of("refreshedLocksSize", response.getRefreshedTokens().size()),
+                SafeArg.of("refreshedAll", request.get().equals(response.getRefreshedTokens())));
```

Comment:
This is good! But I think we should wrap these calls in an `isTraceEnabled`: in a sense, people do sometimes fire off big lock requests, and `equals(...)` I imagine might not be cheap in some cases so we might not want to do it unless trace logging is turned on

Reply:
Updated